pkg/collector/corechecks/system/winkmem/winkmem.go

```diff
@@ -0,0 +1,239 @@
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"golang.org/x/sys/windows"
+)
+
+const (
+	kmemCheckName = "winkmem"
+
+	// KMemDefaultTopNum is the default number of kernel memory tags to return
+	KMemDefaultTopNum = 10
+)
+
+var (
+	modntdll                     = windows.NewLazyDLL("ntdll.dll")
+	procNtQuerySystemInformation = modntdll.NewProc("NtQuerySystemInformation")
+)
+
+// Config is the configuration options for this check
+// it is exported so that the yaml parser can read it.
+type Config struct {
+	TopPagedBytes                int      `yaml:"topPagedBytes"`
+	TopNonPagedBytes             int      `yaml:"topNonPagedBytes"`
+	TopPagedAllocsOutstanding    int      `yaml:"topPagedAllocsOutstanding"`
+	TopNonPagedAllocsOutstanding int      `yaml:"topNonPagedAllocsOutstanding"`
+	SpecificTags                 []string `yaml:"alloctags"`
+}
+
+// KMemCheck is the agent object for this check.
+type KMemCheck struct {
+	core.CheckBase
+	config Config
+}
+
+func init() {
+	core.RegisterCheck(kmemCheckName, winkmemFactory)
+}
+
+func winkmemFactory() check.Check {
+	return &KMemCheck{
+		CheckBase: core.NewCheckBase(kmemCheckName),
+	}
+}
+
+/*
+configuration:
+
+init_config:
+  topPagedBytes: 10  // indicates record the top 10 paged-pool byte users
+  topNonPagedBytes: 9 // indicates record the top 9 non-paged pool byte users
+  topPagedAllocsOutstanding: 8 // indicates record the top 8 by number of non-freed allocs
+  topNonPagedAllocsOutstanding: 7 // indicates record the top 7 by number of non freed allocs
+  allocTags:
+    - HTab      // indicates record the allocs with memtag `HTab` regardless of whether it's in the top N above
+*/
+
+// Configure is called to configure the object prior to the first run
+func (w *KMemCheck) Configure(data integration.Data, initConfig integration.Data, source string) error {
+
+	// check to make sure the function is actually there, so we can fail gracefully
+	// if it's not
+	if err := modntdll.Load(); err != nil {
+		return err
+	}
+	if err := procNtQuerySystemInformation.Find(); err != nil {
+		return err
+	}
+
+	if err := w.CommonConfigure(data, source); err != nil {
+		return err
+	}
+	cf := Config{
+		TopNonPagedBytes:             KMemDefaultTopNum,
+		TopPagedBytes:                KMemDefaultTopNum,
+		TopPagedAllocsOutstanding:    KMemDefaultTopNum,
+		TopNonPagedAllocsOutstanding: KMemDefaultTopNum,
+	}
+	err := yaml.Unmarshal(initConfig, &cf)
+	if err != nil {
+		return err
+	}
+
+	w.config = cf
+	log.Infof("Winkmem config %v", w.config)
+	return nil
+}
+
+// Run executes the check
+func (w *KMemCheck) Run() error {
+	sender, err := aggregator.GetSender(w.ID())
+	if err != nil {
+		return err
+	}
+	spi, err := getpoolinfo()
+	if err != nil {
+		return err
+	}
+	// make a map to keep track of the indexes we actually want
+	// to send up (so we don't double count them)
+	tagmap := make(map[int]bool)
```

Comment:
Looks like this is actually a set. You may want to consider using `struct{}` as a map value. This will allow you to remove `if v {` condition down below.